crates/core/src/parameters.rs

```diff
@@ -12,6 +12,46 @@ use super::hash::Hash;
 use super::time::DurationSecs;
 use super::token;
 use crate::borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+use crate::storage;
+
+/// Abstract parameters storage keys interface
+pub trait Keys {
+    /// Key for implicit VP
+    fn implicit_vp_key() -> storage::Key;
+}
+
+/// Abstract parameters storage read interface
+pub trait Read<S> {
+    /// Storage error
+    type Err;
```

Comment:
yeah, I was thinking the same - maybe we could put these into a new crate `namada_systems` or something - It would actually help to have a concrete type for the error too to avoid having to type `Err = namada_state::StorageError` in some places

Reply:
I've done this in a follow-up PR #3472